src/Adapter/MSTest.TestAdapter/Discovery/AssemblyEnumerator.cs

```diff
@@ -278,13 +297,82 @@ private bool DynamicDataAttached(IDictionary<string, object?> sourceLevelParamet
         // If you remove this line and acceptance tests still pass you are okay.
         test.TestMethod.DataType = DynamicDataType.None;
 
-        // NOTE: From this place we don't have any path that would let the user write a message on the TestContext and we don't do
-        // anything with what would be printed anyway so we can simply use a simple StringWriter.
-        using var writer = new StringWriter();
-        TestMethod testMethod = test.TestMethod;
-        MSTestAdapter.PlatformServices.Interface.ITestContext testContext = PlatformServiceProvider.Instance.GetTestContext(testMethod, writer, sourceLevelParameters);
-        TestMethodInfo? testMethodInfo = _typeCache.GetTestMethodInfo(testMethod, testContext, MSTestSettings.CurrentSettings.CaptureDebugTraces);
-        return testMethodInfo != null && TryProcessTestDataSourceTests(test, testMethodInfo, tests);
+        return testMethodInfo.Value != null && TryProcessTestDataSourceTests(test, testMethodInfo.Value, tests);
+    }
+
+    private static void AddFixtureTests(TestMethodInfo testMethodInfo, List<UnitTestElement> tests, HashSet<string> fixtureTests)
+    {
+        string assemblyName = testMethodInfo.Parent.Parent.Assembly.GetName().Name!;
+        string assemblyLocation = testMethodInfo.Parent.Parent.Assembly.Location;
+        string className = testMethodInfo.Parent.ClassType.Name;
+        string classFullName = testMethodInfo.Parent.ClassType.FullName!;
+
+        if (!fixtureTests.Contains(assemblyLocation))
+        {
+            _ = fixtureTests.Add(assemblyLocation);
+
+            if (testMethodInfo.Parent.Parent.AssemblyInitializeMethod is not null)
+            {
+                tests.Add(GetAssemblyFixtureTest(testMethodInfo.Parent.Parent.AssemblyInitializeMethod, assemblyName, className, classFullName, assemblyLocation, Constants.AssemblyInitializeFixtureTrait));
```

Comment:
Might be better to use an enum than strings for the fixture type.

Reply:
I was thinking about it. Strings are more flexible for the display purpose. [Assembly Fixture @ Microsoft].AssemblyInitialize. This can be done using [Description] attribute but it adds some complexity to the code.
Also, in hierarchy we need to pass string so there will be enum to string conversion over there.